internal/controllers/machineset/machineset_controller.go

```diff
@@ -676,14 +675,8 @@ func (r *Reconciler) updateExternalObject(ctx context.Context, obj client.Object
 	updatedObject.SetLabels(machineLabelsFromMachineSet(machineSet))
 	updatedObject.SetAnnotations(machineAnnotationsFromMachineSet(machineSet))
 
-	patchOptions := []client.PatchOption{
-		client.ForceOwnership,
-		client.FieldOwner(machineSetManagerName),
-	}
-	if err := r.Client.Patch(ctx, updatedObject, client.Apply, patchOptions...); err != nil {
-		return errors.Wrapf(err, "failed to update %s", klog.KObj(obj))
-	}
-	return nil
+	_, err := ssa.Run(ctx, r.Client, r.ssaRequestTracker, obj, updatedObject, machineSetManagerName)
```

Comment:
Just curious: is there a reason why this Reconciler method doesn't return the updated object like the others above?

Reply:
Basically because we don't need the resulting object.

For this reconcile we just write the InfraMachine and the BootstrapConfig and then we don't need them later.

It's different for Machine.